phys2bids/interfaces/txt.py

```diff
@@ -14,6 +14,42 @@
 LGR = logging.getLogger(__name__)
 
 
+def multifreq(timeseries, freq):
+    """
+    Checks if there are channels with different frequency than the maximum one
+
+    Parameters
+    ----------
+    timeseries: list
+        list with channels only in np array format
+    freq : list
+        list with the maximun frequency
+
+    Returns
+    -------
+    mfreq: list
+        new list with the actual frequency of the channels
+    """
+    mfreq = []
+    # for each channel check frequency
+    for idx, chann in enumerate(timeseries):
+        eq_samples = 1  # start counter
+        # get value on the channel
+        for idx2, value in enumerate(chann[1:]):
+            # if value equal to previous value
+            if value == chann[idx2]:
+                # sample duplicated by interpolation
+                # increase counter so we have the number of
+                # equal samples continuos
+                eq_samples += 1
+            else:
+                break
```

Comment:
@vinferrer , do you think we can check that the same amount of repeated samples is itself repeated twice or three times? Now, if for any reason a channel is perfectly 0 (say, the TR trigger) for the first 20 seconds, but then starts changing, we risk to save that channel with a different frequency than the real one. If we check that the "interpolation" is something recursive a couple of times, we're safe to assume is due to frequency resample.

Reply:
@smoia, last commit changed the function, now instead of checking only the begginning, the function asks for a start and a leftout. The function takes a part of the channels indicated by these parameters and in the while loop creates a list that contains how many times a number was repeated until changing. From this list the function counts the ocurrences and gets the most repeated number. This is the number that the frequency is divided by in the end. Theoretically the function could cover the hole channel, however I use the start and leftout arguments to make it faster.

Reply:
resolved